### ISSUE TITLE
WriteFile ignores Close() error

### ISSUE THREAD

**Issue by kevinburke** [dchest/safefile#7]
Maybe it's fine, but `defer f.Close()` can end up ignoring an error if f.Close() returns one. Maybe this is fine since we're already returning an error if f.Close() returns a non-nil err and it would be hard to return two errors at once.

**Comment by dchest** [comment]
Yes, it's fine, `Close` can be safely called without error checking if you do `Commit` call and check its error. `WriteFile` returns the result of `Commit`.

See this comment on `Close` function:

```
// If the file has been committed, Close is no-op.
```

https://github.com/dchest/safefile/blob/master/safefile.go#L102

(If the file was not committed, you'll get the error from `Commit`).

Thanks for code review anyway!

**Comment by dchest** [comment]
Actually, you're right: we won't detect a failure to remove file if renaming fails. https://github.com/dchest/safefile/blob/master/safefile.go#L124

**Comment by dchest** [comment]
... ah, but in this case, we still get the actual error indicating that rename failed. So, there will be an error anyway 🤷‍♂️
